pkg/api/v1/types.go

```diff
@@ -710,6 +725,22 @@ type Container struct {
 	TTY   bool `json:"tty,omitempty" description:"Whether this container should allocate a TTY for itself, also requires 'stdin' to be true; default is false"`
 }
 
+func (ctr *Container) ApplyDefaults() {
```

Comment:
I have to admit that I'm with @bgrant0607 here, I'd prefer to see us take advantage of Go's tag system to do the defaulting, so the flow would be something like:

   * Add tags to struct fields needing defaults
   * Use reflection and code-gen to generate the defaulting code
   * Inject it back in, similar to parts of this PR...